### PR TITLE
Make Preview and Save Draft buttons use the same style

### PR DESCRIPTION
**Pull request by ZebulanStanphill**
## Description
We were moving away from showing chevrons in buttons that open dropdown menus... so why does the Preview button still have one? This PR removes it.

One issue this raises to the surface is the inconsistency in design between the "Save Draft" and "Preview" buttons. I'm not really sure what an icon-less button _should_ look like. What exactly is the standard?

**UPDATE:** this PR now changes both buttons to look the same and adjusts their padding to optically balance the spacing between them.

## Screenshots 
image
image

**UPDATE:** the buttons now look like this:
image
image

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jasmussen** [comment]
Interesting! And thank you for the pull request. 

A PR like this is in the camp of "we can try this and learn"! Trying it out in code is a much better way to progress, than talking endlessly on tickets. In that vein, I wouldn't oppse getting a feel for this. 

However my instinct is that this control has more in common with the dropdown than it does with icon button menu toggles:

77938587-f54d7d80-7283-11ea-82ea-5ebab3b914ab

I know there's a spectrum of consistency there, but in my mind it's the same design principle as having a solid button component that can also be presented in outline or text-only form. More than ensuring consistency that buttons all look the same, we leverage our good sensibilities to decide which presentation works best given its context.

Yes, we can use that same principle to decide that in the top toolbar, that preview button does not need an icon. But to me it's more about evaluating what we feel works best there, than ensuring consistency.

@pablohoneyhoney @mtias any thoughts?

**Comment by ZebulanStanphill** [comment]
@jasmussen Yeah, I made this PR since I figured it would be easier to try it out in practice than just discussing it in an issue.

I'd argue this button is less like a select dropdown than a lot of other things, e.g. the block alignment toolbar dropdown button. The preview options button includes not only a 3-choice radio menu, but also an action link to open something in another tab. That makes it a lot less like a select dropdown than, say, the Tools (edit/select) dropdown that's also in the top toolbar.

**Comment by mapk** [comment]
Great questions being raised here! The Preview option works exactly the same as the ellipses on the end. It opens a dropdown that shows some editor settings which change the view immediately when selected.

Screen Shot 2020-05-06 at 7 41 07 AM

There seems to be two major differences:

1. The Preview is limited to one type of setting change.
2. The Preview is in word form, not an icon.

So how do we navigate the UI implementation with those observations? Maybe because the focus of the Preview is limited to one type of selection, it should remain as a pseudo selectbox component (ie. include the dropdown arrow)?

**Comment by ZebulanStanphill** [comment]
@mapk
>The Preview is limited to one type of setting change.

Actually, it also has the "Preview in external tab" link, though I suppose that's not really a "setting". But even ignoring that, it still wouldn't be the only dropdown in the top toolbar with a single type of setting change:

image

So then the only real distinguishing factor is that it is a text-only button, rather than an icon button.

**Comment by mapk** [comment]
Good point, @ZebulanStanphill. I noted here https://github.com/WordPress/gutenberg/issues/20877#issuecomment-624813197, which is an attempt to add a navigation ability to the top toolbar, that including a dropdown arrow would help visually.

I wasn't thinking about this particular issue, but if the similarity is that these two items are text-based and not icons... that starts aligning nicely.

**Comment by ZebulanStanphill** [comment]
Well, the site editor things ended up _not_ using chevrons. Notably, the site editor navigation buttons use the same styles as the preview button in this PR, so it looks like the "Save Draft" button is the one that is inconsistent.

So, how should we move forward here?

**Comment by jasmussen** [comment]
@ZebulanStanphill good observations. I think next step is a PR if you're up for it? No better way to decide.

**Comment by ZebulanStanphill** [comment]
I could definitely make a PR to simplify the Save Draft button.

I just looked into it, though, and it turns out that button is actually not using custom styles... it's using **tertiary** button styles, specifically the `is-tertiary` CSS class. Personally, I think that's really weird, because for something that's supposed to be "tertiary", it has a much stronger hover state than the Preview button. :confused:

Do you still think I should make a PR to change it to a standard button?

**Comment by jasmussen** [comment]
> Personally, I think that's really weird, because for something that's supposed to be "tertiary", it has a much stronger hover state than the Preview button. 😕

I'm increasingly unsure that the terms primary, secondary, and tertiary to define the visual appearance of the buttons is not really helpful, because those terms suggest a hierarchy that is not always appropriate for the design. For example in embeds, buttons go primary, tertiary, tertiary. I think it might make sense to consider them as solid, outlined, and text-only buttons. 

But that is a separate point, and does not detract from yours, and it does seem like the next thing to try is to make both the save draft button, and the preview dropdown, be the same as they are in the FSE experience. It might even be an extension, or refocus, of this PR. Thanks!

**Comment by ZebulanStanphill** [comment]
>But that is a separate point, and does not detract from yours, and it does seem like the next thing to try is to make both the save draft button, and the preview dropdown, be the same as they are in the FSE experience.

The Save Draft button isn't in the FSE experience, and the Preview dropdown button looks the same as in the post/page editor.

But anyway, I've updated the PR to change both buttons to appear consistent with other text-only buttons like the ones in the FSE experience.
image

**Comment by jasmussen** [comment]
Here's what I see:

Screenshot 2020-06-05 at 08 15 15

As an exercise in reduction, this seems very promising, insofar as it surfaces inconsistencies that were previously overlooked. For example it immediately stands out to me that the titlecase, "Save Draft" is inconsistent with what we're moving towards elsewhere. We might want to change that to "Save draft".

I also wonder if we should unify the padding around this text only button to match tertiary buttons. Compare:

Screenshot 2020-06-05 at 08 15 50

Screenshot 2020-06-05 at 08 15 55

The padding of tertiary buttons is 6px compared to 6px 12px for primary and secondary buttons, so that the optical spacing of the buttons when not focused is balanced. 

Here's what these tweaks would look like:

Screenshot 2020-06-05 at 08 16 44

Here's focus:

padding

I think this could work.

**Comment by ZebulanStanphill** [comment]
@jasmussen That sounded like a good idea to me, so I started implementing it.

However, while working on it, I noticed something odd: some instances of the `<Button>` component use the `isSecondary` prop, while others don't use it. It looks like you have to use the prop to get the `is-secondary` class applied to the resulting HTML.

It makes me wonder: why do some use the prop and some don't? Are some of them invalid? If the `isSecondary` prop represents a default button (or at least that's what I would have assumed), then why isn't it applied by default? Should I apply the `isSecondary` prop to the button versions of the `PostSavedState` component?

**Comment by ZebulanStanphill** [comment]
Oh wait, apparently `isSecondary` buttons are styled with borders (like the `isTertiary` hover style) by default. So there actually is a difference between `isSecondary` and no class at all. Well that's confusing. :confounded:

**Comment by ZebulanStanphill** [comment]
Well, I'm a bit stuck on trying to adjust the button styles. There's a lot of different places adjusting the same things and I'm not sure how to do this properly.

**Comment by jasmussen** [comment]
Thank you for working on this. This is what I see:

buttons

This is feeling pretty good to me. The reduction is beneficial. We might even remove the checkmark next to "Saved", and make "Saved" the same dark color? Something to explore separately. 

I understand you got stuck here. I took a quick look, and before I knew it, I had created a separate PR: #22986 — that PR is 90% your work, and if the code in that PR checks out, feel free to rebase this PR off that, so you can get the merge props. 

I think the trick was to lean into the isTertiary prop. Combined with that, there was some bad CSS to remove here and there. 

Nice work.

**Comment by jasmussen** [comment]
I think this is worth trying 👍

**Comment by ZebulanStanphill** [comment]
Alright, this PR should be ready for final review and merge. :slightly_smiling_face:

With regard to the "Saved" state of the save draft button, I think that in another PR, we should get rid of that state entirely and replace it with a snackbar notification, similar to the one shown upon publish.

**Comment by jasmussen** [comment]
> With regard to the "Saved" state of the save draft button, I think that in another PR, we should get rid of that state entirely and replace it with a snackbar notification, similar to the one shown upon publish.

Definitely agree on separate exploration. I think we'll want to think through the whole flow, though, as no label currently means the document is new. Could be a non issue in practice, just want to make sure we catch any quirks.

**Comment by ZebulanStanphill** [comment]
Fixed failing unit test.

**Comment by shaunandrews** [comment]
This looks good.

I do wonder why the padding on the left and right of the save and preview buttons is smaller than the publish button? The publish button uses 12px of padding, while these buttons now use 6px.

Here's how it looks now in this PR:

6px-padding

Here's how it would look if you made it 12px to match the publish button:

12px-padding

--

I wouldn't let this hold up merging, but it wasn't clear to me if this was intentional. (Sorry if I missed a previous discussion around this decision.)

**Comment by ZebulanStanphill** [comment]
@shaunandrews The padding was changed to optically balance the spacing between the buttons (similar to what is done to the buttons in the rich text formatting tools toolbar group) and match existing tertiary buttons that already do the same thing. @jasmussen first [suggested it 4 days ago](https://github.com/WordPress/gutenberg/pull/21192#issuecomment-639283523).

**Comment by jasmussen** [comment]
Yep, the reduced padding for tertiary buttons is intentional, and you have to compare the resting state rather than the focused state, as that is what you will see most. It's also the case with the media placeholders:

Screenshot 2020-06-09 at 09 08 45

It becomes especially important when stacked:

Screenshot 2020-06-09 at 09 09 36

Even that isn't perfect, but it's better than the alternative of a 12px indented stacked button.

**Comment by shaunandrews** [comment]
I tend to disagree. The smaller padding looks wrong to my eyes, especially with the example of the vertical setup state for the image block; with different padding the text doesn’t line up. 

That said, I doubt anyone will really notice and this is such a small thing that it’s not worth discussing more now. Thanks for the insights.

**Comment by shaunandrews** [comment]
I just noticed that when I create a new document, I see this disabled preview button:

image

**Comment by ZebulanStanphill** [comment]
For better or worse, that's what the standard disabled state of tertiary buttons looks like. If we want to change it, we should probably do so in a separate PR.

**Comment by youknowriad** [comment]
What's blocking here? I feel this is a good one to land before 5.5

**Comment by jasmussen** [comment]
I had my checks pass today by updating snapshots. Normaly it's enough to update the unit test snapshots:

`npm run test-unit -- --updateSnapshot`

but for my last two PRS, I had to do 

`npm run test-e2e -- -u`

that last one took forever, but fixed the issue I was having.

**Review by jasmussen — APPROVED** [review]
Before:

Screenshot 2020-06-23 at 08 49 18

Screenshot 2020-06-23 at 08 49 23

After:

after

Screenshot 2020-06-23 at 08 48 42

This is a small change, but it adds a lot of noise reduction, at what does not feel like it's at the expense of anything. It also adds consistency with label case, and colors with buttons in the block toolbar:

Screenshot 2020-06-23 at 08 49 03

I think we should get this in.

**Comment by ZebulanStanphill** [comment]
I'll merge once I can get the tests to pass.

**Pull request merged**
